# The original fai-cd uses the aufs filesystem.  We cannot use the aufs filesystem on Docker (which uses aufs).
# This patch also removes any .svn subdirectories from the image and makes it USB stick bootable.
# This patch was tested on version 4.3.3
--- /usr/sbin/fai-cd.orig	2015-10-02 12:40:46.518758426 -0500
+++ /usr/sbin/fai-cd	2015-10-02 12:41:38.670758697 -0500
@@ -179,9 +179,8 @@
         umount $nfsrootdir/$d 2>/dev/null
     done
     mountpoint -q $nfsrootdir && umount $nfsrootdir
-    mountpoint -q $cow && umount $cow
     set -e
-    rm -rf $tmp $cow
+    rm -rf $tmp
 }
 # - - - - - - - - - - - - - - - - - - - - - - - - - -
 sleep_now() {
@@ -195,8 +194,8 @@
 
     echo "Writing FAI CD-ROM image to $isoname. This may need some time."
     xorriso -report_about HINT -as mkisofs -graft-points -b $boot_image \
-        -V "$vname" -A "$aname" \
-        -no-emul-boot -boot-info-table --embedded-boot ${embed_image} --protective-msdos-label -o $isoname \
+        -V "$vname" -A "$aname" -m .svn \
+        -no-emul-boot -boot-info-table --embedded-boot ${embed_image} -partition_offset 16 -no-pad -o $isoname \
         -R $tmp --sort-weight 0 / --sort-weight 1 /boot || die 12 "xorriso failed."
     echo -n "ISO image size and filename: "; du -h $isoname
 }
@@ -234,22 +233,6 @@
     
     mount --bind $mirrordir $nfsrootdir/media/mirror && echo "Mirror $mirrordir mounted"
     mount -o remount,ro,bind $nfsrootdir/media/mirror
-# TODO: customize /etc/apt, copy apt preferences etc.
-
-    # create the sources.list for the CD
-    cat > $nfsrootdir/etc/apt/sources.list <<EOF
-# mirror location for fai CD, file generated by fai-cd
-EOF
-
-    dists=$(find $mirrordir -name "Packages*" | grep binary | sed 's/binary-.*//' | \
-         sed "s#$mirrordir/*dists/##" | xargs -r -n 1 dirname | sort | uniq )
-    [ -z "$dists" ] && die 19 "No suitable Packages file found in mirror."
-
-    for i in $dists ; do
-        comp=$(find $mirrordir/dists/$i -maxdepth 2 -type d -name "binary-*" | \
-        sed -e "s#$mirrordir/*dists/$i/##" -e 's#/binary-.*##' | uniq | tr '\n' " ")
-        echo "deb file:/media/mirror $i $comp" >> $nfsrootdir/etc/apt/sources.list
-    done
 
 }
 # - - - - - - - - - - - - - - - - - - - - - - - - - -
@@ -345,17 +328,15 @@
 fi
 
 tmp=$(mktemp -t -d fai-cd.XXXXXX) || exit 13
-cow=$(mktemp -t -d cow-XXXXXX)
-mount -n -t tmpfs tmpfs $cow
 
 
 # check if dracut or live-boot is used
 if [ -d $ONFSROOT/live ]; then
     hidedirs="$hidedirs /boot"
     if [ $bootonly -ne 1 ]; then
-	mkdir -p $tmp/live/filesystem.dir
+	mkdir -p $tmp/live
 	nfsrootdir=$tmp/live/filesystem.dir
-	mount -t aufs -o noatime,noxino,dirs=$cow=rw:$ONFSROOT/live/filesystem.dir=rr aufs $nfsrootdir && echo "NFSROOT $NFSROOT mounted"
+	cp -a $ONFSROOT/live/filesystem.dir $nfsrootdir
     fi
 else
     # things to do for dracut
@@ -378,7 +359,6 @@
 
 if [ $bootonly -eq 1 ]; then
     mkiso
-    mountpoint -q $cow && umount $cow
 else
     trap "unhide_dirs;umount_dirs" EXIT ERR
     create_iso
